enhancements/network/egress-ip-multi-nic.md

```diff
@@ -0,0 +1,287 @@
+inet6 fe80::21df:c25a:dc63:f6eb/64 scope link noprefixroute
+valid_lft forever preferred_lft forever
+```
+
+Cluster manager would assign the `EgressIP` to worker 2.
+
+To allow packets to be steered into secondary networks, OVN reroute logical policies on `ovn_cluster_router` will redirect
+the packet out to the management port of the node which the egress IP is hosted.
+
+We will add rules in linux routing policy database control for each source Pod IP and lookup the correct routing table
+which will contain one route. This single route within a new managed routing table will direct the packet to the correct interface.
+
+We will also add a masquerade rule to iptables when a packet leaving a particular interface matches the source pod IP.
+
+
+See section `Design Details` for full implementation details.
+
+### User Stories
+As a user, I want the ability for cluster workload traffic to be routed to a network that is not managed by OVN and to have a configurable source IP.
+
+### Test Plan
+E2E tests verifying the correct source IP for deployment modes global IC and multi-zone IC:
+- Day 2 Networks CRUD including mask resizing
+- Configure new EgressIP both on ovn managed and unmanaged networks
+- Migrate existing EgressIP on ovn managed and unmanaged networks
+- Delete existing EgressIP on ovn managed and unmanaged networks
+- Add node
+- Delete node
+- Edit node
+- Add pod
+- Delete pod
+- Remove pod
+- Delete a valid rule manually 
+- Delete a valid IPtables rule manually
+- Delete a valid route manually
+- Delete a valid egress IP assigned to an interface manually
+- Valid egress service and egress IP functionality if both are active on the same k8 objs
+Today, if egress service and egress IP both 'match' on the same pod, egress service is preferred
+Continue this for this enhancement where egress service and egress IP overlap, prefer egress service
+- Egress IP interface is now a child interface and attached to a bond which is now the IP of the EIP interface. Reconcile to move the EIP to the bond.
+- Egress IP interface is now a child interface and attached to a bond which is now a different IP of the original EIP interface. Reconcile to move the EIP
+to a different network and if none found, clear EIP node status and let cluster manager reconcile and find a new node
+- Egress IP balancing of non-ovn EIP assignments is the same as current EIP balancing
+- Mixing Egress IPs from ovn and non-ovn managed networks
+- GARPs are sent when egress IP moves from a non-ovn managed network
+- Link goes down
+- EIP removed
+- Original network that allowed EIP assignment is removed
+
+### API Extensions
+Add `Network` field to `EgressIP` `.status`:
+
+```golang
+// The per node status, for those egress IPs who have been assigned.
+type EgressIPStatusItem struct {
+// Assigned node name
+Node string `json:"node"`
+// Assigned egress IP
+EgressIP string `json:"egressIP"`
+// Assigned network
+Network string `json:"network"`
```

Comment:
What will we put here as the a value? The name of the interface? I would assume only cluster-manager can update this so it would need to determine the value from the host-addresses annotation?

Reply:
It is not the name of an interface but a network (i.e. `10.10.10.4/24`).
It will be a network found in `host-addresses` annotation.
Correct, only CM can set it.
Nothing will depend on this value though to allow safe upgrades.
If any other component needs to calculate the network the can.
This is mainly for debug.